bats/tests/profile/sample.bats

```diff
@@ -32,69 +45,188 @@ start_app() {
+    PROFILE_TYPE=$PROFILE_DEFAULTS
     add_profile_bool application.startInBackground "$PROFILE_START_IN_BACKGROUND"
+    add_profile_string containerEngine.name "$PROFILE_CONTAINER_ENGINE"
+    add_profile_string kubernetes.version "$PROFILE_DEFAULTS_KUBERNETES_VERSION"
 }
 
-@test 'start app with new profile' {
+@test 'add settings to locked profile' {
+    PROFILE_TYPE="$PROFILE_LOCKED"
+    add_profile_bool containerEngine.allowedImages.enabled "$PROFILE_USE_IMAGE_ALLOW_LIST"
+    add_profile_list containerEngine.allowedImages.patterns "${PROFILE_IMAGE_PATTERNS[@]}"
+    add_profile_bool application.extensions.allowed.enabled "$PROFILE_USE_EXTENSION_LIST"
+    add_profile_list application.extensions.allowed.list "${PROFILE_EXTENSION_LIST[@]}"
+    add_profile_string kubernetes.version "$PROFILE_LOCKED_KUBERNETES_VERSION"
+}
+
+@test 'start app with new profiles' {
     RD_CONTAINER_ENGINE=""
     start_app
 }
 
-@test 'verify profile settings' {
+@test 'verify profile settings were applied' {
+    verify_settings
+}
+
+@test 'try to install a not-allowed extension' {
+    run rdctl extension install ignatandrei/blockly-automation
+    assert_failure || return
+    run rdctl extension install joycelin79/newman-extension:0.0.5
+    assert_failure || return
+}
+
+@test 'install an allowed extension' {
+    run rdctl extension install joycelin79/newman-extension:0.0.7
+    assert_success || return
+}
+
+@test 'try to change locked fields via rdctl set' {
+    run rdctl set --container-engine.allowed-images.enabled=false
+    assert_failure || return
+
+    run rdctl set --kubernetes.version="1.16.15"
+    assert_failure || return
+}
+
+@test 'try to change locked fields via API' {
+    run rdctl api /v1/settings -X PUT --body "{\"version\": \"$RD_API_VERSION\", \"containerEngine\": {\"allowedImages\": { \"patterns\": [ \"pattern1\" ] }}}"
+    assert_failure || return
+    run rdctl api /v1/settings -X PUT --body "{\"version\": \"$RD_API_VERSION\", \"containerEngine\": {\"allowedImages\": {\"enabled\": false }}}"
+    assert_failure || return
+    run rdctl api /v1/settings -X PUT --body "{\"version\": \"$RD_API_VERSION\", \"application\": {\"extensions\": { \"allowed\": false }}}"
+    assert_failure || return
+    run rdctl api /v1/settings -X PUT --body "{\"version\": \"$RD_API_VERSION\", \"application\": {\"extensions\": { \"list\": [\"pattern1\"] }}}"
+    assert_failure || return
+    run rdctl api /v1/settings -X PUT --body "{\"version\": \"$RD_API_VERSION\", \"kubernetes\": {\"version\": \"1.16.15\"}}"
+    assert_failure || return
+}
+
+@test 'ensure locked settings are preserved' {
     verify_settings
 }
 
 @test 'change defaults profile setting' {
-    rdctl set --application.start-in-background=false
+    run rdctl set --application.start-in-background=false
+    assert_success || return
+    run rdctl set --application.auto-start=true
+    assert_success || return
+    run rdctl set --kubernetes.version=1.19.16
```

Comment:
You are changing the k8s version here, but I don't see any tests that verify that the change has been applied.

Also the version should probably be a constant defined at the top of the test, so we can quickly see the different versions being needed for this test.

Reply:
The change in kubernetes version is being addressed by `verify_settings` , where after we try to set a random kubernetes version, we run verify_settings to make sure the locked version is in place, with an additional check that the default version did not override the locked version. Did not test the random version explicitly, but if `assert` succeeds in verifying the version is the locked one, this means the random version was indeed not applied